internal/types/resource/resource.go

```diff
@@ -90,3 +93,17 @@ var Map = map[string]Type{
 	CredentialStore.String():   CredentialStore,
 	CredentialLibrary.String(): CredentialLibrary,
 }
+
+type MinimalInfo struct {
+	PublicId string
+	ScopeId  string
+	UserId   string
+}
+
+type MinimalInfoRepo interface {
```

Comment:
The names of this struct and this interface could be revisited. At first read it isn't clear what they are and what their purpose is. And perhaps the package they are in adds to the difficulty of understanding what they are.